spark/Debugging-extended-Spark-History-Server.md

```diff
@@ -0,0 +1,202 @@
+Extended Spark History Server is built on top of the community version of Spark History Server and provide advanced features. The current extension includes data tab and graph tab. Data tab shows the input/output data source of the Spark job including table operations. This allows users to validate the output, preview input schema/data to facilitate the development. Graph tab provides job graph, playback and heatmap. This helps users to understand the bottleneck, monitor and diagonose issues. Please check [Extended Spark UI documentation](https://go.microsoft.com/fwlink/?linkid=2005401) for detail information.
+
+## 1. Hotfix extended Spark History Server
+
+To hotfix with new version, use the script below which will download and update spark-enhancement.jar*.
+
+**upgrade_spark_enhancement.sh**:
+
+   ```bash
+    #!/usr/bin/env bash
+
+    # Copyright (C) Microsoft Corporation. All rights reserved.
+
+    # Arguments:
+    # $1 Enhancement jar path
+
+    if [ "$#" -ne 1 ]; then
+        >&2 echo "Please provide the upgrade jar path."
+        exit 1
+    fi
+
+    install_jar() {
+        tmp_jar_path="/tmp/spark-enhancement-hotfix-$( date +%s )"
+
+        if wget -O "$tmp_jar_path" "$2"; then
+            for FILE in "$1"/spark-enhancement*.jar
+            do
+                back_up_path="$FILE.original.$( date +%s )"
+                echo "Back up $FILE to $back_up_path"
+                mv "$FILE" "$back_up_path"
+                echo "Copy the hotfix jar file from $tmp_jar_path   to $FILE"
+                cp "$tmp_jar_path" "$FILE"
+
+                "Hotfix done."
+                break
+            done
+        else    
+            >&2 echo "Download jar file failed."
+            exit 1
+        fi
+    }
+
+    jars_folder="/usr/hdp/current/spark2-client/jars"
+    jar_path=$1
+
+    if ls ${jars_folder}/spark-enhancement*.jar 1>/dev/null 2>&1;   then
+        install_jar "$jars_folder" "$jar_path"
+    else
+        >&2 echo "There is no target jar on this node. Exit with no action."
+        exit 0
+    fi
+   ```
+
+**Usage**: 
+
+`upgrade_spark_enhancement.sh https://${jar_path}`
+
+**Example**:
+
+`upgrade_spark_enhancement.sh https://${account_name}.blob.core.windows.net/packages/jars/spark-enhancement-${version}.tgz` 
+
+**To run the bash file from Azure portal**
+
+1. Launch [Azure Portal](https://ms.portal.azure.com), and select your cluster.
+2. Click **Script actions**, then **Submit new**. Complete the **Submit script action** form, then click **Create** button.
+    
+    + **Script type**: select **Custom**.
+    + **Name**: specify a script name.
+    + **Bash script URI**: upload the bash file to private cluster then copy URL here. Alternatively, use the URI provided.
+    
+   ```upgrade_spark_enhancement
+    https://hdinsighttoolingstorage.blob.core.windows.net/shsscriptactions/upgrade_spark_enhancement.sh
```

Comment:
Update this to github repo where script action is uploaded like - https://raw.githubusercontent.com/hdinsight/script-actions/master/install-spark2-3/install-spark2-3-v04.sh

Reply:
Same with above, please review this PR https://github.com/hdinsight/script-actions/pull/13/files

Reply:
i meant update the url to point to this github source instead of blob store

Reply:
Yes, need this to be checked in to generate the URL to update here:) https://github.com/hdinsight/script-actions/pull/13